File: com/editor/editor.go
package editor

import "github.com/lincaiyong/page/com"

func Editor() *Component {
	ret := &Component{}
	ret.BaseComponent = com.NewBaseComponent[Component]("div", ret)
	return ret
}

type Component struct {
	*com.BaseComponent[Component]
	onCreated com.Method
	_destroy  com.Method
	setValue  com.Method
}

File: com/editor/editor.js
function _destroy() {
    this._editor.dispose();
    super._destroy();
}

function setValue(v) {
    this._editor.setValue(v);
}

function setLanguage(v) {
    monaco.editor.setModelLanguage(this._editor.getModel(), v);
}

function onCreated() {
    const options = {
        value: '',
        language: '',
        theme: 'vs',
        automaticLayout: true,
        lineNumbers: 'on', // 'off'
        minimap: {
            enabled: false,
        },
        readOnly: true,
        // fontFamily: '',
        // glyphMargin: false,
        // suggestOnTriggerCharacters: false,
    };
    this._editor = monaco.editor.create(this.ref, options);
}